src/main/java/ua/ithillel/lesson8/FileNavigator.java

```diff
@@ -0,0 +1,120 @@
+package ua.ithillel.lesson8;
+
+import java.util.*;
+
+public class FileNavigator {
+
+    private final List<FileData> totalList = new ArrayList<>();
+    private final Map<String, List<FileData>> map = new HashMap<>();
+
+    public Map<String, List<FileData>> getMap() {
+        return map;
+    }
+
+    public void add(FileData fileData){
+        String currentPath = fileData.getPath();
+
+        for (Map.Entry<String, List<FileData>> me: map.entrySet()){
+            if(currentPath.equals(me.getKey())){
+                me.getValue().add(fileData);
+                return;
+            }
+        }
+
+        List<FileData> list = new ArrayList<>();
+        list.add(fileData);
+        map.put(currentPath, list);
+    }
+
+    public List<FileData> find(String path){
+
+        for (String pathKey: map.keySet()){
+            if(path.equals(pathKey)){
+                return map.get(pathKey);
+            }
+        }
+        return null;
+    }
+
+    public List<FileData> filterBySize(int maxSize){
+
+        List<FileData> limitSizeList = new ArrayList<>();
+
+        for (Map.Entry<String, List<FileData>> me: map.entrySet()){
```

Comment:
```suggestion
        for (List<List<FileData>> me: map.values()){
```
ключі тобі не потрібні тут, достатньо по значенням пройтися.